app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -6,6 +6,7 @@ class Ovirt < ComputeResource
     validates :url, :format => { :with => URI.regexp }
     validates :user, :password, :presence => true
     before_create :update_public_key
+    before_save :update_available_operating_systems
```

Comment:
I've made sure the before_validatation is not called on referencing objects (tested with host with compute resource association). So I've updated it here.